dicom-archive/dicomTar.pl

```diff
@@ -137,18 +137,19 @@
 my $creator         = $summary->{user};
 my $sumTypeVersion  = $summary->{sumTypeVersion}; 
 
-my $byDate;
 # Determine how to name the archive... by acquisition date or by today's date.
-if ($todayDate) { $byDate = $today; } else { $byDate = $summary->{header}->{scandate}; } # wrap up the archive 
-my $finalTarget = "$targetlocation/DCM_${byDate}_$summary->{metaname}.tar";
+my $byDate      = $todayDate ? $today : $summary->{header}->{scandate};
+my $pname       = $summary->{header}->{pname};
```

Comment:
Do you know if patient names can contain special characters, like `!`, `%`, etc...Might be worth doing

`my $pname = quotemeta($summary->{header}->{pname};`

in order to 100% sure that the file name generated with this is a valid Unix file name. But then this might be a bit paranoid...Up to you.

Reply:
Good catch! It is unlikely but we never know. Adding the quotemeta function